Title: IBM Error captured in Process Task Manager when starting Case Foundation Case Analyzer: "The Event Dispatch Manager cannot start a thread because there is a problem with the objectstore null". - United States

Text:
 TECHNOTE (TROUBLESHOOTING)

PROBLEM(ABSTRACT)
 Error captured in Process Task Manager when starting Case Foundation Case Analyzer: "The Event Dispatch Manager cannot start a thread because there is a problem with the objectstore null. See the chained exceptions for more information. Verify that the object store can be accessed and it is 
enabled for Advanced Case Management". 

SYMPTOM
Error captured in Process Task Manager when starting Case Analyzer.


The Event Dispatch Manager cannot start a thread because there is a

problem with the objectstore null. See the chained exceptions for more

information. Verify that the object store can be accessed and it is

enabled for Advanced Case Management.

Cannot access the ObjectStore ERT_Human_Resources. See the chained

exceptions for more information.

java.security.cert.CertPathBuilderException: unable to find valid

certification path to requested target

at filenet.pa.acm.datacollector.CEConfigurationInfoAgent.

getACMObjectStore(Unknown Source)

at filenet.pa.datacollector.EventDispatchManager.start(Unknown

Source)

at filenet.pa.main.PARunnable.run(Unknown Source)

at java.lang.Thread.run(Thread.java:736)

Caused by:

com.filenet.api.exception.EngineRuntimeException: FNRCA0031E:

API_UNABLE_TO_USE_CONNECTION: The URI for server communication cannot

be determined from the connection object: https://P8Demo.ibm.com:9444/wsi/FNCEWS40MTOM/. 



at com.filenet.apiimpl.wsi.ServiceSessionNst.

cOpenMultipartOutputStream(ServiceSessionNst.java:239)

at com.filenet.apiimpl.wsi.ServiceSessionNst.cReqRespPath

(ServiceSessionNst.java:134)

at com.filenet.apiimpl.wsi.ServiceSessionNst.getObjects

(ServiceSessionNst.java:104)

at com.filenet.apiimpl.util.SessionHandle.getObjects

(SessionHandle.java:381)

at com.filenet.apiimpl.core.Session.callGetObjects(Session.java:

90)

at com.filenet.apiimpl.core.Session.executeGetObject(Session.

java:295)

at com.filenet.apiimpl.core.Session.getObject(Session.java:309)

at com.filenet.apiimpl.core.IndependentObjectImpl.getObject

(IndependentObjectImpl.java:149)

at com.filenet.apiimpl.core.IndependentObjectImpl.refresh

(IndependentObjectImpl.java:157)

at com.filenet.api.core.Factory$Domain.fetchInstance(Factory.

java:1652)

... 4 more

Caused by:

javax.net.ssl.SSLHandshakeException: com.ibm.jsse2.util.g: PKIX

path building failed: java.security.cert.CertPathBuilderException:

unable to find valid certification path to requested target

at com.ibm.jsse2.n.a(n.java:36)

at com.ibm.jsse2.sc.a(sc.java:442)

at com.ibm.jsse2.gb.a(gb.java:18)

at com.ibm.jsse2.gb.a(gb.java:205)

at com.ibm.jsse2.hb.a(hb.java:46)

at com.ibm.jsse2.hb.a(hb.java:57)

at com.ibm.jsse2.gb.n(gb.java:285)

at com.ibm.jsse2.gb.a(gb.java:146)

at com.ibm.jsse2.sc.a(sc.java:88)

at com.ibm.jsse2.sc.g(sc.java:437)

at com.ibm.jsse2.sc.a(sc.java:544)

at com.ibm.jsse2.sc.startHandshake(sc.java:124)

at com.ibm.net.ssl.www2.protocol.https.c.afterConnect(c.java:48)

at com.ibm.net.ssl.www2.protocol.https.d.connect(d.java:39)

at com.ibm.net.ssl.www2.protocol.https.b.connect(b.java:83)

at com.filenet.apiimpl.wsi.ServiceSessionNst$1.run

(ServiceSessionNst.java:351)

at java.security.AccessController.doPrivileged(AccessController.

java:251)

at com.filenet.apiimpl.wsi.ServiceSessionNst.

cDoPrivilegedHttpConnectAndGetOutputStream(ServiceSessionNst.java:360)

at com.filenet.apiimpl.wsi.ServiceSessionNst.cInitHttpConnection

(ServiceSessionNst.java:321)

at com.filenet.apiimpl.wsi.ServiceSessionNst.

cOpenMultipartOutputStream(ServiceSessionNst.java:231)

... 13 more

Caused by:

com.ibm.jsse2.util.g: PKIX path building failed: java.security.cert.

CertPathBuilderException: unable to find valid certification path to

requested target

at com.ibm.jsse2.util.e.b(e.java:101)

at com.ibm.jsse2.util.e.b(e.java:35)

at com.ibm.jsse2.util.d.a(d.java:12)

at com.ibm.jsse2.gc.a(gc.java:28)

at com.ibm.jsse2.gc.checkServerTrusted(gc.java:16)

at com.ibm.jsse2.gc.b(gc.java:82)

at com.ibm.jsse2.hb.a(hb.java:308)

... 28 more

Caused by:

java.security.cert.CertPathBuilderException: unable to find valid

certification path to requested target

at com.ibm.security.cert.PKIXCertPathBuilderImpl.buildCertPath

(PKIXCertPathBuilderImpl.java:642)

at com.ibm.security.cert.PKIXCertPathBuilderImpl.engineBuild

(PKIXCertPathBuilderImpl.java:357)

at java.security.cert.CertPathBuilder.build(CertPathBuilder.

java:258)

at com.ibm.jsse2.util.e.b(e.java:49)

... 34 more


CAUSE
SSL certificate was not imported into the Case Analyzer keystore.


RESOLVING THE PROBLEM
Import SSL Certificate in to Case Analyzer java keystore using the keytool command:
<Drive>:\IBM\FileNet\Case Analyzer Engine\java\jre\lib\security 

\CaseAnalyzerInstallation_directory\java\jre\bin\keytool 
-import -file cert_file_name
-alias alias -keystore cacerts -storepass password